Closure_Front_End/src/components/SearchBar.vue

```diff
@@ -48,9 +50,13 @@ export default {
    * more on the implementation of the search on the backend part of the project.
    */ {
     search() {
-      this.$http.get(this.url + this.query)
+      this.$http
+        .get(this.url + this.query, {
+          validateStatus: Utils.validateStatusCode(status)
+        })
         .then((response) => (this.suggestions = response.data.results))
-        .then((this.hide = !this.suggestions.length || this.query == ""));
+        .then((this.hide = !this.suggestions.length || this.query == ""))
+        .catch (Utils.popOutStatusCodeError);
```

Comment:
Manually adding a 'catch' and 'validateStatus' to every `$http` call leads to a lot of duplication

I recommend putting all axios related error handling in a single point, which should also be tied to a vue component for error handling.
See [this](https://stackoverflow.com/questions/48990632/how-to-manage-axios-errors-globally-or-from-one-point) - an example of using an [interceptor](https://axios-http.com/docs/interceptors) to automatically handle errors. This can be done to the axios instance created at 
https://github.com/ScDor/Closure/blob/a48ac99eb8d9958dce73c84bbeb373798c9e538c/Closure_Front_End/src/auth/index.js#L108

(Similarly to how I set the base URL, one can also set `validateStatus` as explained [here](https://axios-http.com/docs/req_config))

Reply:
To clarify, there may be cases where we want to use a `.catch` to deal with specific types of errors that should be handled differently. For example, a business logic error (e.g, adding a non existent course, conflicting semester, etc..) should present a much more informative error UI, that could for example highlight the problematic course, this is in contrast to generic network errors or internal errors that would be dealt by an showing a toast